### ISSUE TITLE
Core dumped when launching a modded minecraft version 1.16 with OpenJ9 Nightly Build

### ISSUE THREAD

**Issue by 1a2s3d4f1** [eclipse-openj9/openj9#14330]
Java -version output
--------------------

```
openjdk version "17.0.1-internal" 2021-10-19
OpenJDK Runtime Environment (build 17.0.1-internal+0-adhoc.jenkins.buildjdk17x86-64windowsnightly)
Eclipse OpenJ9 VM (build master-4ba73dfa870, JRE 17 Windows 7 amd64-64-Bit Compressed References 20220119_140 (JIT enabled, AOT enabled)
OpenJ9   - 4ba73dfa870
OMR      - b128a66
JCL      - 4e273c2c2a6 based on jdk-17.0.1+12)
```

Summary of problem
------------------

I try to launch minecraft with openj9 nightly build,But JVM crash when game finished loading resources.

Diagnostic files
----------------
https://mega.nz/file/SMdGBD6K#_xuBWjc-WUS6FuVpIwLZHSeFyYP0_AjLb4z2r4RctIk

**Comment by pshipton** [comment]
It crashed in the `OpenGL(R) Driver for Intel(R) Graphics Accelerator` library when called from the opengl JNI code. You could try updating the driver. There is also `-Xcheck:jni` which may help identify problems with JNI code, but not sure if this would be helpful in this case. I wonder if something has been garbage collected which the native code didn't expect to be garbage collected. You could also try adding a java.lang.ref.Reference.reachabilityFence() on any such Object after the native call.

Module: C:\Windows\system32\ig7icd64.dll
```
4XESTACKTRACE                at org/lwjgl/opengl/GL20C.glLinkProgram(Native Method)
4XESTACKTRACE                at org/lwjgl/opengl/GL20.glLinkProgram(GL20.java:408)
4XESTACKTRACE                at net/minecraft/class_4493.method_22051(class_4493.java:360)
```

**Comment by 1a2s3d4f1** [comment]
Core dump when using -Xshareclasses:none and -Xcheck:jni:warn
https://mega.nz/file/OdF0UJTb#ycEM0hNiDRLKpRFCQMH7cQuYdWJxzdPKSUkCDy5aGwU
The game crash when I add -Xcheak:jni This is crash report:https://pastebin.com/nkA016JP

**Comment by pshipton** [comment]
The crash report shows
```
Caused by: java.lang.NullPointerException: Cannot invoke "jdk.internal.misc.Unsafe.putLong(java.lang.Object, long, long)" because "sun.misc.Unsafe.theInternalUnsafe" is null
	at jdk.unsupported/sun.misc.Unsafe.putLong(Unsafe.java:273)
	at org.lwjgl.system.MemoryUtil.memPutAddress(MemoryUtil.java:1714)
	at org.lwjgl.system.ThreadLocalUtil.setFunctionMissingAddresses(ThreadLocalUtil.java:199)
	at org.lwjgl.opengl.GL.create(GL.java:209)
```
The crash, this time in the vm, shows the same stack.
```
4XESTACKTRACE                at jdk/internal/misc/Unsafe.putLong(Native Method)
4XESTACKTRACE                at sun/misc/Unsafe.putLong(Unsafe.java:273)
4XESTACKTRACE                at org/lwjgl/system/MemoryUtil.memPutAddress(MemoryUtil.java:1714)
4XESTACKTRACE                at org/lwjgl/system/ThreadLocalUtil.setFunctionMissingAddresses(ThreadLocalUtil.java:199)
4XESTACKTRACE                at org/lwjgl/opengl/GL.create(GL.java:209)
```
@tajila

**Comment by tajila** [comment]
The exception doesnt make sense

```
> !stackslots 0x02a26f00
<2a26f00> *** BEGIN STACK WALK, flags = 00400001 walkThread = 0x0000000002A26F00 ***
<2a26f00> 	ITERATE_O_SLOTS
<2a26f00> 	RECORD_BYTECODE_PC_OFFSET
<2a26f00> Initial values: walkSP = 0x0000000002B15D80, PC = 0x0000000000000003, literals = 0x0000000000000000, A0 = 0x0000000002B15DD0, j2iFrame = 0x0000000000000000, ELS = 0x0000000001CCE650, decomp = 0x0000000000000000
<2a26f00> INL native method frame: bp = 0x0000000002B15DA0, sp = 0x0000000002B15D80, pc = 0x0000000000000003, cp = 0x0000000000000000, arg0EA = 0x0000000002B15DD0, flags = 0x0000000000000000
<2a26f00> 	Method: jdk/internal/misc/Unsafe.putLong(Ljava/lang/Object;JJ)V !j9method 0x0000000002A521B0
<2a26f00> 	Using signature mapper
<2a26f00> 	Arguments starting at 0x0000000002B15DD0 for UDATA(0x0000000000000006) slots
<2a26f00> 		O-Slot: a0[0x0000000002B15DD0] = 0x00000000DA8D4910 // <--sun.misc.Unsafe.theInternalUnsafe
<2a26f00> 		O-Slot: a1[0x0000000002B15DC8] = 0x0000000000000000 //obj arg
<2a26f00> 		I-Slot: a2[0x0000000002B15DC0] = 0x00000000E4580AD0 //long arg
<2a26f00> 		I-Slot: a3[0x0000000002B15DB8] = 0x000007FEED3B1B98
<2a26f00> 		I-Slot: a4[0x0000000002B15DB0] = 0x00000AD80000073B //2nd long arg
<2a26f00> 		I-Slot: a5[0x0000000002B15DA8] = 0x000000001CAB0840
<2a26f00> Bytecode frame: bp = 0x0000000002B15DE8, sp = 0x0000000002B15DD8, pc = 0x000000001D4CA55F, cp = 0x0000000003C9E180, arg0EA = 0x0000000002B15E18, flags = 0x0000000000000000
<2a26f00> 	Method: sun/misc/Unsafe.putLong(Ljava/lang/Object;JJ)V !j9method 0x0000000003C9D7F8
<2a26f00> 	Bytecode index = 7
<2a26f00> 	Using local mapper
<2a26f00> 	Locals starting at 0x0000000002B15E18 for 0x0000000000000006 slots
<2a26f00> 		I-Slot: a0[0x0000000002B15E18] = 0x00000000DC4AB828
<2a26f00> 		I-Slot: a1[0x0000000002B15E10] = 0x0000000000000000
<2a26f00> 		I-Slot: a2[0x0000000002B15E08] = 0x00000000E458CFC0
<2a26f00> 		I-Slot: a3[0x0000000002B15E00] = 0x000007FEED3B1B98
<2a26f00> 		I-Slot: a4[0x0000000002B15DF8] = 0x000000001D5DBCED
<2a26f00> 		I-Slot: a5[0x0000000002B15DF0] = 0x000000001CAB0840
```

The receiver

```
> !j9object 0x00000000DA8D4910
!J9Object 0x00000000DA8D4910 {
	struct J9Class* clazz = !j9class 0x2A57200 // jdk/internal/misc/Unsafe
	Object flags = 0x00000000;
	I lockword = 0x00000000 (offset = 0) (java/lang/Object) <hidden>
}
```

**Comment by tajila** [comment]
@1a2s3d4f1 does it fail when you run with -Xint?

**Comment by 1a2s3d4f1** [comment]
@tajila It fall with -Xint

**Comment by tajila** [comment]
@1a2s3d4f1 No not necessary. Can you give me instructions on how to reproduce this? I have a linux x86-64 env, can it work there?

**Comment by 1a2s3d4f1** [comment]
@tajila Running vanilla minecraft with openj9 nightly build can reproduce this issue. It happend on vanilla minecraft 1.16.5 or 1.18.1. I use [HMCL](https://github.com/huanghongxun/HMCL/releases) to launch game.

**Comment by 1a2s3d4f1** [comment]
@tajila Core dump not happen on Linux envoroment(VirsualBox). Maybe it only happen on windows system.

**Comment by 1a2s3d4f1** [comment]
@tajila Core dump not happen on Linux envoroment(VirsualBox). Maybe it only happen on windows system.

**Comment by tajila** [comment]
@mikezhang1234567890 Please take a look at this.

Try to reproduce with `-Xint -XX:+DebugInterpreter` so you can get the most up to date stack frame

**Comment by 1a2s3d4f1** [comment]
It only happend on this computer. Another computer(Not inter graphics) not dump.
New core dump(Added option"-Xint -XX:+DebugInterpreter" ):
https://mega.nz/file/WJ9EFYRL#8ahRgRzxz3OEzGt6SHFkFt1EMsmGq6b623w2BHHORvM

**Comment by 1a2s3d4f1** [comment]
It fixed on 0.32 version, so close this issue.
